src/unix/notbsd/mod.rs

```diff
@@ -53,7 +53,9 @@ s! {
         pub ai_protocol: ::c_int,
         pub ai_addrlen: socklen_t,
 
-        #[cfg(any(target_os = "linux", target_os = "emscripten"))]
+        #[cfg(any(target_os = "linux",
+                  target_os = "emscripten",
```

Comment:
So it turns out that doesn't work, because the structure is serialized, and order matters. Reverted.